### ISSUE TITLE
Mark some properties in the all.yaml file as required

### ISSUE THREAD

**Issue by ognis1205** [unitycatalog/unitycatalog#280]
**Describe the bug**

I'm working on converting [the OpenAPI specification](https://github.com/unitycatalog/unitycatalog/blob/main/api/all.yaml) to TypeScript interfaces. I have found some properties are nullable while they should be (potentially) required, e.g., I guess the `name` and `catalog_name` of `SchemaInfo` should be required:

https://github.com/unitycatalog/unitycatalog/blob/main/api/all.yaml#L995

**To Reproduce**

I'm using [openapi-typescript](https://www.npmjs.com/package/openapi-typescript) to convert the specification.

1. npx openapi-typescript /path/to/spec/api.yaml --output /path/to/output/schema.ts

**Expected behavior**

Some output properties should not be `Optional` in a typed language like TypeScript.

**Additional context**

Basically, it’s not that problematic, but when using a typed language, it gets converted to Optional (in TypeScript, fields with a ? suffix), which can be a bit cumbersome during type checking and type guarding. It would be helpful if the OpenAPI spec could include required fields. 

<img width="541" alt="Screenshot 2024-07-25 at 13 11 59" src="https://github.com/user-attachments/assets/7ad3c1af-1c5a-4dba-bcf1-5084aef95f7d">
<img width="529" alt="Screenshot 2024-07-25 at 13 12 19" src="https://github.com/user-attachments/assets/1fd9a3f2-b107-407f-90a4-fcb0ea323f39">

**Comment by ognis1205** [comment]
@dennyglee @tdas

**Comment by dennyglee** [comment]
Thanks @ognis1205  - would love you to tackle this PR per Slack convo :)

**Comment by MrPowers** [comment]
Thanks for opening this issue @ognis1205
